bukkit/src/main/java/us/myles/ViaVersion/bukkit/handlers/BukkitDecodeHandler.java

```diff
@@ -69,10 +68,6 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf bytebuf, List<Object> l
             // call minecraft decoder
             try {
                 list.addAll(PipelineUtil.callDecode(this.minecraftDecoder, ctx, bytebuf));
-            } catch (InvocationTargetException e) {
```

Comment:
I agree. Doing this breaks it more when the exception is actually thrown

Reply:
> 
> 
> Breaks behaviour (no longer assessing underlying exception)

Because underlying exception will not be thrown due direct access to encode/decode methods.
Reflection is no longer used

Reply:
that's not what the exception is for. That partciular one might not be thrown, but I'd suggest you read up on https://docs.oracle.com/javase/7/docs/api/java/lang/reflect/InvocationTargetException.html
and https://docs.oracle.com/javase/7/docs/api/java/lang/invoke/MethodHandle.html#invoke(java.lang.Object...)

Reply:
I know for what this exceptions was "made", but I still don't get what's the problem

Reply:
the catch block you removed

Reply:
Yes, and? I removed it so the main handler can catch the exception
My code is equivalent to previous one.
try {} catch (e) { throw e; }

Reply:
Also previous code ignored and type of error that does not extend `Exception`. OOM/any others does not.

Reply:
rip me for the two points on that ex I guess, overlooked that 🤦‍♂️